generatorcore/eb_to_bisko/__init__.py

```diff
@@ -0,0 +1,862 @@
+from dataclasses import dataclass
+
+from generatorcore.inputs import Inputs
+
+from ..utils import div
+from .. import (
+    agri2018,
+    electricity2018,
+    business2018,
+    fuels2018,
+    heat2018,
+    industry2018,
+    lulucf2018,
+    residences2018,
+    transport2018,
+)
+
+
+@dataclass
+class EnergySourceCalcIntermediate:
+
+    eb_energy_from_same_sector: float
+
+    eb_CO2e_cb_from_same_sector: float | None = None
+
+    eb_energy_from_agri: float | None = None
+
+    eb_CO2e_cb_from_heat: float | None = None
+    eb_CO2e_cb_from_elec: float | None = None
+    eb_CO2e_cb_from_fuels: float | None = None
+    eb_CO2e_cb_from_agri: float | None = None
+
+    energy: float = 0
+    CO2e_cb: float = 0
+
+    # production based Emissions
+    CO2e_pb: float | None = None
+
+    eb_CO2e_pb_from_heat: float | None = None
+
+    def __post_init__(self):
+        def sum_over_none_and_float(*args: float | None) -> float:
+            return_float: float = 0
+            for elem in args:
+                if elem is None:
+                    continue
+                return_float += elem
+
+            return return_float
+
+        # automatically sum over all cb_based emissions
+        self.CO2e_cb = sum_over_none_and_float(
+            self.eb_CO2e_cb_from_same_sector,
+            self.eb_CO2e_cb_from_heat,
+            self.eb_CO2e_cb_from_elec,
+            self.eb_CO2e_cb_from_fuels,
+            self.eb_CO2e_cb_from_agri,
+        )
+        self.energy = sum_over_none_and_float(
+            self.eb_energy_from_same_sector, self.eb_energy_from_agri
+        )
+        self.CO2e_pb = self.eb_CO2e_pb_from_heat
+
+
+@dataclass
+class SumClass:
```

Comment:
Maybe call this Sums instead of SumClass ? Or are these particularly classy sums ;-P